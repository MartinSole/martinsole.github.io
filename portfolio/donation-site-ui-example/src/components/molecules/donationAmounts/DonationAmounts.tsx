import { RadioButton } from '../../atoms'

interface Props {
    donationAmounts: number[]
}


const DonationAmounts = ({donationAmounts}: Props) => {
  return (
    <fieldset className='donationAmountsContainer gridContainer gridGap'>
      <legend className="srOnly">Choose an amount to donate</legend>
        {donationAmounts.map((donationAmount, index) => <RadioButton {...{
          id: `donation-amount-${index}`,
          name: 'donation-amount',
          text:`Â£${donationAmount}`,
          index,
        }}
        key={donationAmount}
          />)}
    </fieldset>
  )
}

export default DonationAmounts