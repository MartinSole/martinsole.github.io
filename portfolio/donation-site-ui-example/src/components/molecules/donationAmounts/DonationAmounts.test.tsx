import '@testing-library/jest-dom'
import { render } from "@testing-library/react"
import DonationAmounts from "./DonationAmounts"

test("Renders the DonationAmounts component", () => {
    const mockProps = {
        donationAmounts: [1,2,3,4,5]
    }
   const {container} = render(<DonationAmounts donationAmounts={mockProps.donationAmounts}/>)
    expect(container).toMatchSnapshot()
    const radioButtons = container.querySelectorAll('input')
    expect(radioButtons).toHaveLength(5)
})
