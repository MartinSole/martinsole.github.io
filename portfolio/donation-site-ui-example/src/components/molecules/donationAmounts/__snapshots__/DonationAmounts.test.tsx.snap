// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the DonationAmounts component 1`] = `
<div>
  <fieldset
    class="donationAmountsContainer gridContainer gridGap"
  >
    <legend
      class="srOnly"
    >
      Choose an amount to donate
    </legend>
    <input
      checked=""
      id="donation-amount-0"
      name="donation-amount"
      type="radio"
      value="£1"
    />
    <label
      class="buttonBase"
      for="donation-amount-0"
    >
      £1
    </label>
    <input
      id="donation-amount-1"
      name="donation-amount"
      type="radio"
      value="£2"
    />
    <label
      class="buttonBase"
      for="donation-amount-1"
    >
      £2
    </label>
    <input
      id="donation-amount-2"
      name="donation-amount"
      type="radio"
      value="£3"
    />
    <label
      class="buttonBase"
      for="donation-amount-2"
    >
      £3
    </label>
    <input
      id="donation-amount-3"
      name="donation-amount"
      type="radio"
      value="£4"
    />
    <label
      class="buttonBase"
      for="donation-amount-3"
    >
      £4
    </label>
    <input
      id="donation-amount-4"
      name="donation-amount"
      type="radio"
      value="£5"
    />
    <label
      class="buttonBase"
      for="donation-amount-4"
    >
      £5
    </label>
  </fieldset>
</div>
`;
